### ISSUE TITLE
Screen frozen if push to TabNavigation from a TabNavigation's second tab then pop using default backBehavior

### ISSUE THREAD

**Issue by kolyneh** [react-navigation/react-navigation#4626]
#### Given navigation schema:
    
    --Stack 
      |--TabNav 1
      |       |-- Tab A
      |       |-- Tab B
      |--TabNav 2
              |-- Tab C
              |-- Tab D
    
 #### Steps
  1. Start `TabNav 1`, navigate to `Tab B`,
  2. Push to `TabNav 2`, navigate to `Tab D`, 
  3. Press back button twice, the screen pop up to `TabNav 1`,
  4. But `TabNav 1` shows `Tab A`, instead of `Tab B`, and now the screen is frozen! 

- What code are you running and what is happening?
````

const HomeTabNav = createBottomTabNavigator({
        Device: {
            screen:  DeviceScreen,
        },
        Group: {
            screen: GroupScreen,
        },
    }, {
        tabBarOptions: {
            activeTintColor: color.common_accent_color,
            inactiveTintColor: color.common_primary_color,
            style: {
                backgroundColor: color.main_tab_bg,
            },
        },
    }
   );

HomeTabNav.navigationOptions = {
    header: null,   
};


const ControlTabNav = createBottomTabNavigator({
    AdjustTab: {
        screen: ControlPage,
    },
    SceneTab: {
        screen: ScenePage,
    },
}, {
    tabBarOptions: {
        activeTintColor: color.common_accent_color,
        inactiveTintColor: color.common_primary_color,
        style: {
            backgroundColor: color.main_tab_bg,
        },
    },
});


const AppStack = createStackNavigator({
    Home: {
        screen: HomeTabNav,
    },
    Control: ControlTabNav,
});

````

    
If add `backBehavior:'none'` to the `HomeTabNav`, the screen will not freeze.

But there's question:
Why would `HomeTabNav` handles `backPress` event when it's not on the top stack and then freeze?
    
### Expected Behavior

- What do you expect should be happening?
  Screen is not frozen, and the previous TabNavigation should show the second screen using the default 
 `backBehavior`.

### How to reproduce
[https://snack.expo.io/H1USdoFfX](url)

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation | 2.5.5
| react-native     | 0.55.4
| node             | 8.9.4
| npm   | 5.6.0

**Comment by brentvatne** [comment]
hello! the code you have provided is not runnable. please post a link to an example on http://snack.expo.io/ that i can run and i can look into this

**Comment by kolyneh** [comment]
@brentvatne The code is added.

**Comment by brentvatne** [comment]
thank you! this is fixed in 2.6.0 now
